src/server/clientiface.cpp

```diff
@@ -339,6 +339,12 @@ void RemoteClient::GetNextBlocks (
 				if (d >= d_opt && block->isAir())
 						continue;
 			}
+			if ((!block || !block->isGenerated()) && emerge->isBlockInQueue(p)) {
```

Comment:
one thing I would do is
```suggestion
			const bool want_emerge = !block || !block->isGenerated();
			if (want_emerge && emerge->isBlockInQueue(p)) {
```
and then also use this variable further below

Reply:
Done...